pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -666,14 +684,28 @@ func (p *PriorityQueue) isPodBackingoff(podInfo *framework.QueuedPodInfo) bool {
 
 // SchedulingCycle returns current scheduling cycle.
 func (p *PriorityQueue) SchedulingCycle() int64 {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
+	p.activeQLock.RLock()
+	defer p.activeQLock.RUnlock()
 	return p.schedulingCycle
 }
 
+func (p *PriorityQueue) dumpClusterEvents(inFlightPod *list.Element) []*clusterEvent {
+	var events []*clusterEvent
+	for event := inFlightPod.Next(); event != nil; event = event.Next() {
+		e, ok := event.Value.(*clusterEvent)
+		if !ok {
+			// Must be another in-flight Pod (*v1.Pod). Can be ignored.
+			continue
+		}
+		events = append(events, e)
+	}
+	return events
+}
+
 // determineSchedulingHintForInFlightPod looks at the unschedulable plugins of the given Pod
 // and determines the scheduling hint for this Pod while checking the events that happened during in-flight.
 func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) queueingStrategy {
```

Comment:
actually why do we need activeQLock in this function in the first place?

Reply:
`inFlightPods` and `inFlightEvents` have to be protected using `activeQLock` as they are mutated in the `Pop()` method, which doesn't use the generic `lock`

Reply:
Okay, can you add a comment of which fields should be protected by `activeQLock`, probably above `activeQLock` field definition?

Reply:
Or maybe on each field instead.

```go
// ....
// It should be protected by activeQLock.
inFlightEvents *list.List 

// It should be protected by activeQLock
Hoge xxx
```

Reply:
Good idea, added the comments for the `activeQLock` and all the fields that needs it